lib/policial/limits_checker.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'logger'
+
+module Policial
+  # Public: Checks number of files received from GitHub and notifies if user
+  # reached GitHub per_page limit.
+  class LimitsChecker
+    DEFAULT_PERPAGE = 30
+
+    def initialize(github_client:, files:, logger: ->(text) { puts text })
+      @github_client = github_client
+      @files = files
+      @logger = logger
+    end
+
+    def call
+      return unless @github_client.eql?(Octokit) ||
+                    @github_client.is_a?(Octokit::Client)
```

Comment:
No need for this type checking. The gem blindly assumes that all clients are Octokit clients.